### PR TITLE
Remove redirections to modules moved in v2.0

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This PR removes the backwards compatibility redirections for the `gas_solids_contactors`, `generic_models`, `power_generation` and `surrogates` folders and associated code that has been moved in the new repo structure.

This PR also fixes some imports pointing to the old locations that were missed previously.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by bpaul4 — COMMENTED** [review]
This looks good, thanks for your time and efforts here. It looks to me like the doc error comes from `docs/explanations/modeling_extensions/surrogate/plotting/index.rst`, there are a few places where `automethod` references a deprecated import path.

**Review by eslickj — APPROVED** [review]
Looks good to me although I did not check each file.  I think the examples tests are a subset of the full examples tests.  Should we also make sure all the examples work before merging this?

**Comment by andrewlee94** [comment]
@eslickj Yes, we need to test the examples before this is merged (hence the draft PR and WIP title). Plus, we won;t be merging this until the August release is finished.

File: idaes/models/properties/modular_properties/eos/tests/test_ceos_CO2_H2O.py [hunk 1]
```diff
@@ -1,144 +0,0 @@
-#################################################################################
```
Comment (bpaul4) [inline idaes/models/properties/modular_properties/eos/tests/test_ceos_CO2_H2O.py:1]: Why is this file being removed?

Comment (andrewlee94) [reply idaes/models/properties/modular_properties/eos/tests/test_ceos_CO2_H2O.py:1]: It is being removed as it does not belong here - this is a test depends on a file in `models_extra` and core tests should not depend on non-core models. I think I moved these tests somewhere else.

**Review by bpaul4 — APPROVED** [review]

**Pull request merged**